source/common/config/grpc_subscription_impl.cc

```diff
@@ -79,11 +91,21 @@ void GrpcSubscriptionImpl::onConfigUpdate(
     const std::string& system_version_info) {
   disableInitFetchTimeoutTimer();
   stats_.update_attempt_.inc();
+  auto start = dispatcher_.timeSource().monotonicTime();
   callbacks_.onConfigUpdate(added_resources, removed_resources, system_version_info);
+  std::chrono::milliseconds update_duration = std::chrono::duration_cast<std::chrono::milliseconds>(
+      dispatcher_.timeSource().monotonicTime() - start);
   stats_.update_success_.inc();
   stats_.update_time_.set(DateUtil::nowToMilliseconds(dispatcher_.timeSource()));
   stats_.version_.set(HashUtil::xxHash64(system_version_info));
   stats_.version_text_.set(system_version_info);
+  stats_.update_duration_.recordValue(update_duration.count());
+
+  // TODO(adamsjob): make the threshold configurable, either through runtime or bootstrap
+  if (update_duration.count() > 50) {
+    ENVOY_LOG(debug, "Config update for gRPC config for {} with {} resources took {} ms", type_url_,
+              added_resources.size(), update_duration.count());
```

Comment:
make the log differentiable with the log above? it can also including interested resource names.

Reply:
Done. Changed log wording, added a dump of the resources being updated.